aries_cloudagent/messaging/jsonld/api_schemas.py

```diff
@@ -0,0 +1,97 @@
+"""openapi schemas."""
+from ...messaging.models.openapi import OpenAPISchema
+from marshmallow import Schema, fields, validate
+from marshmallow.validate import Regexp
+
+# https://tools.ietf.org/html/rfc3986#appendix-B
+uri_validate = Regexp(r"^(([^:/?#]+):)?(//([^/?#]*))?([^?#]*)(\?([^#]*))?(#(.*))?")
+# RFC3339 datetime regex taken from https://stackoverflow.com/a/24544212
+datetime_validate = Regexp(
+    rf"^\\d{4}-\\d{2}-\\d{2}T\\d{2}%3A\\d{2}%3A\\"
+    rf"d{2}(?:%2E\\d+)?[A-Z]?(?:[+.-](?:08%3A\\d{2}|\\d{2}[A-Z]))?$"
+)
```

Comment:
Where possible, we should avoid writing our own regex for these values. I think there is already a datetime validator in Marshmallow though it may not be the exact format we're looking for. I'd be surprised if there wasn't one for URIs as well but I'm not sure there is.